Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasedProductsFetcher.swift
+//
+//  Created by Andrés Boedo on 3/17/23.
+
+import Foundation
+import StoreKit
+
+/// A type that can fetch purchased products from StoreKit 2.
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+class PurchasedProductsFetcher {
```

Comment:
Do we need to add tests to this class? (Not sure how difficult it is to mock StoreKit transactions in swift)

Reply:
We can sort of mock transactions by making purchases. We could mock `StoreKit.Transaction.currentEntitlements`, but then we're not really testing anything.
The worry I had is that this might be flaky because `StoreKit.Transaction.currentEntitlements` might return stuff from other tests, but let me see if I can write a test that passes consistently.